### PR TITLE
Lots of fixes and updates to GNSS/GPS plugin

### PR DESCRIPTION
**Pull request by bsimmo**
Bug fixes, timefix, Logging fix, lon/lat order fix, other fixes.  Still one bug to sort. 

Updated to include RMC as ZDA for time is not always sent and GNSS/GPS devices
Latitude generally always comes before Longitude, 
added LOG_LEVEL debug code (best guess at levels)
Sorted out variables surrounding lat/lon (simplified a lot).
Corrected use of raspi-config to newer version.
Other bits and bobs.
time not tested on ZDA fully as I don't have one, will have a go in a few days. It'll be the date part if anything.
Opens up ability to get a date/time variable from GNSS devices (not implemented)

*Still has a bug w.r.t valid log stopping GNSS screen updates on re-entering plugin page*

### CODE REVIEW

File: _P082_GPS.py [hunk 1]
```diff
@@ -7,312 +7,427 @@
...
-            for i, k in enumerate(
-                ['strType', 'fixTime',
-                'lat', 'latDir', 'lon', 'lonDir',
-                'fixQual', 'numSat', 'horDil',
-                'alt', 'altUnit', 'galt', 'galtUnit',
-                'DPGS_updt', 'DPGS_ID']):
-                self.GPSDAT[k] = gpsComponents[i]
+    PLUGIN_ID = 82
+    PLUGIN_NAME = "Position - GNSS/GPS (TESTING)"
+    PLUGIN_VALUENAME1 = "Latitude"
```
Comment (TD-er) [inline _P082_GPS.py:30]: Does changing the order of variables like this render existing code (and rules) incompatible?

**Comment by bsimmo** [comment]
No idea, haven't tested that. Good point.
But the names are still the same and so GNSS#Latitude should still get it
no matter what the variable position is.
Worth a test though.

I'd actually like to have more than 4 available, and make them selectable
but I don't know how.

GNNS devices provide a lot of information and it cannot be used with just
four.

**Comment by TD-er** [comment]
A bit like it is done now in the GPS plugin for ESPEasy you mean :)

**Comment by enesbcs** [comment]
> No idea, haven't tested that. Good point. But the names are still the same and so GNSS#Latitude should still get it no matter what the variable position is. Worth a test though. I'd actually like to have more than 4 available, and make them selectable but I don't know how. GNNS devices provide a lot of information and it cannot be used with just four.

In the Sysinfo plugin, you can see an approach of selectable variables:
https://github.com/enesbcs/rpieasy/blob/master/_P026_Sysinfo.py

**Pull request merged**